base/experiment.go

```diff
@@ -34,19 +33,22 @@ type ExperimentSpec []Task
 
 // Experiment struct containing spec and result
 type Experiment struct {
-	// Tasks is the sequence of tasks that constitute this experiment
-	Tasks ExperimentSpec
+	// Spec is the sequence of tasks that constitute this experiment
+	Spec ExperimentSpec `json:"spec" yaml:"spec"`
 
 	// Result is the current results from this experiment.
 	// The experiment may not have completed in which case results may be partial.
-	Result *ExperimentResult
+	Result *ExperimentResult `json:"result" yaml:"result"`
 
 	// driver enables interacting with experiment result stored externally
 	driver Driver
 }
 
 // ExperimentResult defines the current results from the experiment
 type ExperimentResult struct {
+	// Revision of this experiment
```

Comment:
Idea is to store experiment revision in the experiment result. This is still not fully tested / implemented properly.